ml/src/main/scala/ai/h2o/sparkling/ml/algos/H2OSupervisedAlgorithm.scala

```diff
@@ -46,19 +46,19 @@ abstract class H2OSupervisedAlgorithm[B <: H2OBaseModelBuilder : ClassTag, M <:
     schema
   }
 
-  override protected def preProcessBeforeFit(trainFrameKey: String): Unit = {
-    super.preProcessBeforeFit(trainFrameKey)
-    if (parameters._distribution == DistributionFamily.bernoulli || parameters._distribution == DistributionFamily.multinomial) {
+  override protected def prepareH2OTrainFrameForFitting(trainFrameKey: String): Unit = {
+    super.prepareH2OTrainFrameForFitting(trainFrameKey)
+    val distribution = DistributionFamily.valueOf(getDistribution())
+    if (distribution == DistributionFamily.bernoulli || distribution == DistributionFamily.multinomial) {
       if (RestApiUtils.isRestAPIBased()) {
         val trainFrame = H2OFrame(trainFrameKey)
-        if (trainFrame.columns.find(_.name == getLabelCol()).get.dataType == H2OColumnType.`enum`) {
+        if (trainFrame.columns.find(_.name == getLabelCol()).get.dataType != H2OColumnType.`enum`) {
```

Comment:
Do we really want to change the condition `dataType == H2OColumnType.enum`? What about regression and numeric label?